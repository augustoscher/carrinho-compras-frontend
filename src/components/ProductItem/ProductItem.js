import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Context from '../../context';

const Container = styled.div`
  border: 1px solid #d7dff0;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.04);
  border-radius: 4px;
  padding: 6px 12px;
  background: #fbfcff;
  display: flex;
  margin-bottom: 8px;
`;

const Img = styled.img`
  width: 100px;
  height: 100px;
`;

const Content = styled.div`
  display: flex;
`;

const Slot = styled.div`
  padding: 10px 30px;

  p {
    font-size: 20px;
    font-weight: 500;
    margin-top: 8px;
  }

  span {
    color: #7c7e83;
  }

  button {
    margin: 25px;
    width: 120px;
    height: 30px;
  }
`;

const ProductItem = ({ id, name, photo, price, stock }) => {
  const { dispatch } = useContext(Context);
  const handleClick = () => {
    dispatch({
      type: 'ADD_PRODUCT',
      payload: [{ id, name, photo, price, stock }],
    });
  };
  return (
    <Container>
      <Img src={photo} />
      <Content>
        <Slot>
          <span>Cod: {id}</span>
          <p>
            <strong>{name}</strong>
          </p>
        </Slot>
        <Slot>
          <span>Price:</span>
          <p>
            <strong>{price}</strong>
          </p>
        </Slot>
        <Slot>
          <span>Stock:</span>
          <p>
            <strong>{stock}</strong>
          </p>
        </Slot>
        <Slot>
          <button onClick={handleClick}>Add to Cart</button>
        </Slot>
      </Content>
    </Container>
  );
};

ProductItem.defaultProps = {
  id: '',
  name: '',
  photo: '',
  price: 0,
  stock: 0,
};

ProductItem.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  photo: PropTypes.string,
  price: PropTypes.number,
  stock: PropTypes.number,
};

export default ProductItem;
