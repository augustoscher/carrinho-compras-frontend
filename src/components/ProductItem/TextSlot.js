import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Root = styled.div`
  padding: 10px 30px;

  p {
    font-size: 20px;
    font-weight: 500;
    margin-top: 8px;
  }

  span {
    color: #7c7e83;
  }
`;

const TextSlot = ({ label, text }) => (
  <Root>
    <span>{label}</span>
    <p>
      <strong>{text}</strong>
    </p>
  </Root>
);

TextSlot.defaultProps = {
  label: '',
  text: '',
};

TextSlot.propTypes = {
  label: PropTypes.string,
  text: PropTypes.string,
};

export default TextSlot;
