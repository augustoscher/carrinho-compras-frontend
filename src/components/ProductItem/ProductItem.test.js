import React from 'react';
import { render } from '@testing-library/react';
import ProductItem from './ProductItem';

describe('ProductItem', () => {
  test('match snapshot', () => {
    const { asFragment } = render(<ProductItem />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders button', () => {
    const { getByText } = render(<ProductItem />);

    expect(getByText('Add to Cart')).toBeInTheDocument();
  });
});
