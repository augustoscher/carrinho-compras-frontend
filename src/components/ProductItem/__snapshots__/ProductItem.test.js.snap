// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  border: 1px solid #d7dff0;
  box-shadow: 0px 2px 2px rgba(0,0,0,0.04);
  border-radius: 4px;
  padding: 6px 12px;
  background: #fbfcff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.c1 {
  width: 100px;
  height: 100px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  padding: 10px 30px;
}

.c3 p {
  font-size: 20px;
  font-weight: 500;
  margin-top: 8px;
}

.c3 span {
  color: #7c7e83;
}

.c3 button {
  margin: 25px;
  width: 120px;
  height: 30px;
}

<div
    class="c0"
  >
    <img
      class="c1"
      src=""
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span>
          Cod: 
        </span>
        <p>
          <strong />
        </p>
      </div>
      <div
        class="c3"
      >
        <span>
          Price:
        </span>
        <p>
          <strong>
            0
          </strong>
        </p>
      </div>
      <div
        class="c3"
      >
        <span>
          Stock:
        </span>
        <p>
          <strong>
            0
          </strong>
        </p>
      </div>
      <div
        class="c3"
      >
        <button>
          Add to Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
